src/tests/java/htsjdk/samtools/fastq/FastqRecordTest.java

```diff
@@ -0,0 +1,136 @@
+        final String seqHeaderPrefix = "FAKE0003 Original version has Solexa scores from 62 to -5 inclusive (in that order)";
+        final String seqLine = "ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT";
+        final String qualHeaderPrefix = "";
+        final String qualLine = ";<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+        final FastqRecord fastqRecord = new FastqRecord(seqHeaderPrefix, seqLine, qualHeaderPrefix, qualLine);
+
+        Assert.assertNull(fastqRecord.getBaseQualityHeader());
+
+        Assert.assertEquals(fastqRecord.getReadHeader(), seqHeaderPrefix);
+        Assert.assertEquals(fastqRecord.getBaseQualityString(), qualLine);
+        Assert.assertEquals(fastqRecord.getReadString(), seqLine);
+        Assert.assertNotNull(fastqRecord.toString());//just check not nullness
+        Assert.assertNotEquals(fastqRecord, null);
+        Assert.assertNotEquals(null, fastqRecord);
+        Assert.assertEquals(fastqRecord, fastqRecord);
+        Assert.assertNotEquals(fastqRecord, "fred");
+        Assert.assertNotEquals("fred", fastqRecord);
+        Assert.assertEquals(fastqRecord.length(), seqLine.length());
+    }
+
+    @Test
+    public void testBasicEmptyHeaderPrefix() {
+        final String seqHeaderPrefix = "";
+        final String seqLine = "ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT";
+        final String qualHeaderPrefix = "";
+        final String qualLine = ";<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+        final FastqRecord fastqRecord = new FastqRecord(seqHeaderPrefix, seqLine, qualHeaderPrefix, qualLine);
+        Assert.assertNull(fastqRecord.getReadHeader());
+        Assert.assertNull(fastqRecord.getBaseQualityHeader());
+    }
+
+    @Test
+    public void testCopy() {
+        final String seqHeaderPrefix = "FAKE0003 Original version has Solexa scores from 62 to -5 inclusive (in that order)";
+        final String seqLine = "ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT";
+        final String qualHeaderPrefix = "";
+        final String qualLine = ";<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+        final FastqRecord fastqRecord = new FastqRecord(seqHeaderPrefix, seqLine, qualHeaderPrefix, qualLine);
+        final FastqRecord fastqRecordCopy = new FastqRecord(fastqRecord);
+
+        Assert.assertEquals(fastqRecord, fastqRecordCopy);
+        Assert.assertNotSame(fastqRecord, fastqRecordCopy);
+        Assert.assertSame(fastqRecord.getReadString(), fastqRecordCopy.getReadString());
+        Assert.assertSame(fastqRecord.getBaseQualityString(), fastqRecordCopy.getBaseQualityString());
+        Assert.assertSame(fastqRecord.getBaseQualityHeader(), fastqRecordCopy.getBaseQualityHeader());
+    }
+
+    @Test
+    public void testCopyWithNullSeq() {
+        final String seqHeaderPrefix = "header";
+        final String seqLine = null;
```

Comment:
I think if `seqLine` is `null`, then `qualLine` should be required to be `null`.